enterprise/server/backends/kms/kms.go

```diff
@@ -91,9 +160,71 @@ func (k *KMS) initLocalInsecureKMSClient(ctx context.Context) error {
 	return nil
 }
 
+func loadAWSCreds(file string) (*awscreds.Value, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	rs, err := csv.NewReader(f).ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	if len(rs) != 2 {
+		return nil, status.FailedPreconditionErrorf("credentials file not in valid format (expected 2 rows, got %d)", len(rs))
+	}
+	if len(rs[1]) != 2 {
+		return nil, status.FailedPreconditionErrorf("credential file not in valid format (expected 2 columns, got %d", len(rs[0]))
+	}
+	return &awscreds.Value{
+		AccessKeyID:     rs[1][0],
+		SecretAccessKey: rs[1][1],
+	}, nil
+}
+
+// For some reason, the tink library does not support aws credential files with
+// 2 columns so we manually create the AWS SDK client.
+func crateAWSClientWithCreds(arn *awsKMSARN, credsFile string) (registry.KMSClient, error) {
+	credValue, err := loadAWSCreds(credsFile)
+	if err != nil {
+		return nil, status.UnknownErrorf("could not load credentials file: %s", err)
+	}
+	creds := awscreds.NewStaticCredentialsFromCreds(*credValue)
+	sess, err := awssession.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(arn.region),
+	})
+	if err != nil {
+		return nil, status.UnknownErrorf("could not create session: %s", err)
+	}
+	return awskms.NewClientWithKMS(arn.uriLocationPrefix, awssdkkms.New(sess))
+}
+
 func (k *KMS) clientForURI(uri string) (registry.KMSClient, error) {
 	if strings.HasPrefix(uri, gcpKMSPrefix) && k.gcpClient != nil {
 		return k.gcpClient, nil
+	} else if strings.HasPrefix(uri, awsKMSPrefix) && k.awsClients != nil {
+		arn, err := parseAWSARN(uri)
+		if err != nil {
+			return nil, err
+		}
+		k.awsClientsMu.Lock()
+		defer k.awsClientsMu.Unlock()
+		key := fmt.Sprintf("%s-%s", arn.partition, arn.region)
+		client, ok := k.awsClients[key]
+		if ok {
+			return client, nil
+		}
+		if *awsCredentailsFile != "" {
+			client, err = crateAWSClientWithCreds(arn, *awsCredentailsFile)
+		} else {
+			client, err = awskms.NewClient(arn.uriLocationPrefix)
```

Comment:
what context do we use this case in?

Reply:
It would come into play if you were to run the server locally. It would pick up your AWS credentials from a "well known" location configured by the AWS SDK.

Reply:
Could you add a comment about this

Reply:
Added a note to the flag description.